### PR TITLE
fix(bridge): report wait --gone as gone, not found

### LINKED ISSUE #159: wait --gone reports 'found' on success and 'waiting for' on timeout
`wait --gone` prints "found" on success and "waiting for" on timeout, both of which say the opposite of what the command does.

## Reproduce

```
$ tauri-pilot wait "#already-removed" --gone
✓ found

$ tauri-pilot wait "#permanent-element" --gone --timeout 1000
Error: RPC error: Timeout waiting for #permanent-element
```

The first message says an element was found when the command succeeded precisely because it was absent. The second reads as if the command was waiting for the element to appear.

## Cause

`crates/tauri-plugin-pilot/js/bridge.js` resolves with `{ found: true }` on both branches of the `gone` flag, and the timeout rejection is a single string:

```js
rej(new Error("Timeout waiting for " + (selector || ref)));
```

## Expected

- Success with `--gone`: something like `✓ gone`.
- Timeout with `--gone`: `Timeout waiting for #permanent-element to disappear`.

Purely a messaging fix; the wait logic itself is correct.

### PR DESCRIPTION
**Pull request by mpiton**
## Summary

- `wait --gone` now prints `✓ gone` on success and times out with `Timeout waiting for <selector> to disappear`.
- Appear waits are unchanged (`✓ found` / `Timeout waiting for <selector>`).

## Motivation

Success returned `{ found: true }` (CLI: `✓ found`) and the timeout said `Timeout waiting for <selector>`, both of which describe the opposite of waiting for an element to disappear.

Closes #159

## Changes

- Bridge `waitFor` resolves `{ gone: true }` on the gone path and appends `to disappear` to the timeout.
- CLI `format_text` treats `gone` as a status flag (before `found`) so `{ gone: true }` prints `✓ gone`.
- Node tests for immediate gone, mutation gone, gone timeout, and the appear regressions.
- Changelog updated.

`--json` and MCP now return `{ "gone": true }` instead of `{ "found": true }` for gone success.

## Test Plan

- [x] `cargo test --workspace` passes
- [x] `cargo clippy --workspace -- -D warnings` passes
- [ ] Tested manually with a Tauri app (if applicable)

Also: `node --test crates/tauri-plugin-pilot/js/*.test.mjs` (79 passed).

## Checklist

- [x] No `.unwrap()` outside of tests
- [x] All new files are under 150 lines
- [x] Error handling uses `thiserror` (plugin) or `anyhow` (CLI)
- [x] Commit messages follow conventional commits (`feat:`, `fix:`, `refactor:`, etc.)

## Type

fix

---
## Summary by cubic
Fixes `wait --gone` so success and timeout messages describe waiting for an element to disappear instead of appear, closing #159.

- `wait --gone` now resolves `{ gone: true }` and prints `✓ gone`; timeouts report `Timeout waiting for <selector> to disappear`.
- `--json`, MCP, and JSON-RPC responses change from `{ found: true }` to `{ gone: true }` on gone success — a breaking shape change for consumers parsing those fields.
- Adds behavioral tests for immediate gone, mutation gone, gone timeout, appear via mutation, appear timeout, and the unchanged appear path.

<sup>Written for commit 4e5b8feef2ab1de5b4ff923432442ecbd285826f. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/mpiton/tauri-pilot/pull/182?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

- **Bug Fixes**
  - Fixed `wait --gone` success messages to clearly report when an element has disappeared.
  - Improved timeout messages to distinguish between waiting for an element to appear and waiting for it to disappear.
  - Preserved existing behavior and messaging for standard element-appearance waits.
  - Improved status reporting so disappearance results are shown accurately and consistently.
  - JSON, MCP, and JSON-RPC responses now report successful disappearance waits as `{ gone: true }`.

### CODE REVIEW

File: crates/tauri-plugin-pilot/js/bridge.js [hunk 1]
```diff
@@ -1290,25 +1290,31 @@
...
+        rej(new Error(timeoutMsg));
       }, timeout);
 
       var observer = new MutationObserver(function () {
         var found = check();
         if (!gone && found) {
           observer.disconnect();
           clearTimeout(timer);
-          res({ found: true });
+          res(result);
```
Comment (mpiton) [inline crates/tauri-plugin-pilot/js/bridge.js:1313]: > [!IMPORTANT]
> **🟡 Medium · 🧪 Tests** — Appear-via-mutation wait path has no regression test

This PR rewires the MutationObserver appear branch (`!gone && found`) to `res(result)`, but `bridge.wait.test.mjs` only covers an already-present node and the appear timeout. Swapping that condition, or resolving `{ gone: true }` only here, would leave `wait --selector ".loaded"` hanging until timeout after the node is inserted, and the new suite would still pass.

**Fix:** In `crates/tauri-plugin-pilot/js/bridge.wait.test.mjs`, add a sibling of the gone-mutation test: start with `querySelector` returning null, assign an element, call `fireMutation()`, and assert `{ found: true }`.

<sub>🤖 <b>mpiton-review</b> · tests · confidence 0.82</sub>

File: CHANGELOG.md [hunk 2]
```diff
@@ -22,6 +22,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 
+- `wait --gone` now prints `✓ gone` on success and times out with
+  `Timeout waiting for <selector> to disappear`. Success used to
+  return `{ found: true }` (CLI: `✓ found`) and the timeout said
+  `Timeout waiting for <selector>`, both of which describe the
+  opposite of waiting for an element to disappear. [#159]
```
Comment (mpiton) [inline CHANGELOG.md]: > [!IMPORTANT]
> **🟡 Medium · 📄 Docs** — CHANGELOG omits the wait --gone JSON key change for --json and MCP

This bullet documents the CLI text (`✓ gone`) and the old `{ found: true }` payload, but never states that `--json`, MCP, and JSON-RPC now return `{ gone: true }`. A consumer that parsed `found` on the gone path can treat this as a display-only fix and miss the breaking key change.

**Fix:** Name `{ gone: true }` and the `--json` / MCP / JSON-RPC consumers in this bullet, and say they must read `gone` instead of `found`.

```suggestion
- `wait --gone` now returns `{ gone: true }` (`--json`, MCP, JSON-RPC;
  CLI text: `✓ gone`) and times out with
  `Timeout waiting for <selector> to disappear`. Success used to
  return `{ found: true }` (CLI: `✓ found`) and the timeout said
  `Timeout waiting for <selector>`, both of which describe the
  opposite of waiting for an element to disappear. Callers that
  parsed `found` on the gone path must switch to `gone`. [#159]
```

<sub>🤖 <b>mpiton-review</b> · documentation · confidence 0.9</sub>

**Review by mpiton — COMMENTED** [review]
## 🧨 mpiton-review — PR #182

> [!IMPORTANT]
> 💬 **Review posted** — No blocking issue. 2 findings to consider.

**Actionable comments posted: 2**

| Severity | Count | Meaning |
|:---|---:|:---|
| 🔴 Critical | 0 | ship-stopper: security, data loss, broken behaviour |
| 🟠 High | 0 | must fix before merge |
| 🟡 Medium | 2 | fix now or file a follow-up |
| 🔵 Low | 0 | optional polish |

### 📊 Findings by file

| File | 🔴 | 🟠 | 🟡 | 🔵 |
|:---|---:|---:|---:|---:|
| `crates/tauri-plugin-pilot/js/bridge.js` | 0 | 0 | 1 | 0 |
| `CHANGELOG.md` | 0 | 0 | 1 | 0 |

### 🎯 Start here

1. 🟡 **Appear-via-mutation wait path has no regression test** — `crates/tauri-plugin-pilot/js/bridge.js:1313`
2. 🟡 **CHANGELOG omits the wait --gone JSON key change for --json and MCP** — `CHANGELOG.md:29`

<details>
<summary>🤖 Review setup</summary>

| | |
|:---|:---|
| Commit reviewed | `a1eb3525dc` |
| Base | `main` |
| Files changed | 4 (+135/-11) |
| Languages | javascript, rust |
| Reviewers | documentation, lang-rust, maintainability, product-logic, quality, security, tests |
| Scope | changes in this PR, plus existing code they directly break |

</details>

<sub>Generated by <code>mpiton-review</code>. Reply to a comment to push back — findings are ranked by severity × confidence × reviewer agreement.</sub>

File: CHANGELOG.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (mpiton) [reply CHANGELOG.md]: @cubic-dev-ai

> This entry references `[#159]` but no matching `[#159]: https://github.com/mpiton/tauri-pilot/issues/159` definition was added to the link list at the bottom of `CHANGELOG.md`

Fixed in 4e5b8fe. Added `[#159]: https://github.com/mpiton/tauri-pilot/issues/159` right after the `[#157]` line.

File: crates/tauri-plugin-pilot/js/bridge.js (hunk 1 above)
Comment (mpiton) [reply crates/tauri-plugin-pilot/js/bridge.js:1313]: > Appear-via-mutation wait path has no regression test

Fixed in 4e5b8fe. `bridge.wait.test.mjs` now has "wait reports found after the element is inserted": `querySelector` starts out returning null, the test assigns an element, calls `fireMutation()` and expects `{ found: true }`. I flipped the observer condition to `!gone && !found` to check it bites, and that test timed out at 1000 ms while the other five passed.

File: CHANGELOG.md (hunk 2 above)
Comment (mpiton) [reply CHANGELOG.md]: > CHANGELOG omits the wait --gone JSON key change for --json and MCP

Fixed in 4e5b8fe with the suggested wording. `wait()` in `handler.rs` and `call_app_tool` in `mcp.rs` pass the bridge result through as-is, so `--json`, MCP and JSON-RPC all get `{ gone: true }` now. The bullet tells anyone reading `found` on the gone path to switch to `gone`.
